src/blockchain/Blockchain.ts

```diff
@@ -84,46 +99,160 @@ export default class Blockchain implements IBlockchainApi {
-            log.warn(`Extrinsic execution failed`)
-            log.debug(`Transaction detail: ${JSON.stringify(result, null, 2)}`)
-            const extrinsicError: ExtrinsicError =
-              this.errorHandler.getExtrinsicError(result) || ERROR_UNKNOWN
-
-            log.warn(`Extrinsic error ocurred: ${extrinsicError}`)
-            reject(extrinsicError)
-          }
-          if (result.isFinalized) {
-            resolve(result)
-          } else if (result.isError) {
-            reject(
-              new Error(
-                `Transaction failed with status '${result.status.type}'`
+    try {
+      const nonce: Index = await this.getNonce(identity.address)
+      const signed = identity.signSubmittableExtrinsic(tx, nonce.toHex())
+      log.info(`Submitting ${tx.method} with Nonce ${nonce}`)
+      return new Promise<SubmittableResult>((resolve, reject) => {
+        signed
+          .send((result) => {
+            log.info(`Got tx status '${result.status.type}'`)
+            const { status } = result
+            if (ErrorHandler.extrinsicFailed(result)) {
+              log.warn(`Extrinsic execution failed`)
+              log.debug(
+                `Transaction detail: ${JSON.stringify(result, null, 2)}`
               )
-            )
-          }
-        })
-        .catch((err: Error) => {
-          // just reject with the original tx error from the chain
-          reject(err)
-        })
-    })
+              const extrinsicError: ExtrinsicError =
+                this.errorHandler.getExtrinsicError(result) || ERROR_UNKNOWN
+
+              log.warn(`Extrinsic error ocurred: ${extrinsicError}`)
+              this.resetAccountQueue(identity.address)
+              reject(extrinsicError)
+            }
+            if (result.isFinalized) {
+              this.resetAccountQueue(identity.address)
+              resolve(new SubmittableResult(result))
+            } else if (result.isError) {
+              reject(
+                new Error(`Transaction failed with status '${status.type}'`)
+              )
+            }
+          })
+          .catch((error: Error) => {
+            // just reject with the original tx error from the chain
+            reject(error)
+          })
+      })
+    } catch (error) {
+      this.resetAccountQueue(identity.address)
+      throw error
+    }
   }
 
+  /**
+   * Initiates the Nonce retrieval for the given identity.
+   *
+   * @param accountAddress The address of the identity that we retrieve the nonce for.
+   * @returns Promise of the [[Index]] representation of the Transaction nonce of the identity.
+   *
+   */
   public async getNonce(accountAddress: string): Promise<Index> {
-    return this.api.rpc.system.accountNextIndex(accountAddress)
+    // Initiate nonceRetrieval
+    const unlock = await this.lock(accountAddress)
+    // Await execution of the in handleQueue defined resolve function
+    const nonce = await unlock()
+    return nonce
```

Comment:
I think we could simply check if `accountAddress` is in `this.accountNonces` if not, query next nonce, otherwise increase `this.accountNonces[accountAddress]` by one and return it.

This would not be a [critical section](https://en.wikipedia.org/wiki/Critical_section) because [node.js, v8 and stuff is single threaded](https://stackoverflow.com/questions/5153492/models-of-concurrency-in-nodejs).

Or did I overlook something?